14_functions_lab/05.py

```diff
@@ -1,10 +1,9 @@
-"""
-Write a function groupby that takes a list
-and a function and returns a dictionary
-keyd by the return value of the function on the list items
-
-For example:
-    groupby(lambda s: s[0], ['foo', 'fi', 'hello', 'hi'])
-    returns: { 'f': ['foo','fi'], 'h': ['hello', 'hi'] }
-"""
-
+#Create dictionary from a list of arguments
+from collections import defaultdict
+
+def groupby(func, *params):
+    groupbydict = defaultdict(list)
+    [groupbydict[func(word)].append(word) for word in params]
+    return groupbydict
+
+print groupby(lambda(s): s[0], 'hello', 'hi', 'help', 'bye', 'here')

```

Comment:
Nice, though please note, the example in the original file expected 2 parameters, while your solution expects a function and an unlimited number of individual values. Not a Pythonic list.